File: sdk/typescript/src/2026-04/services/customer_portal/customer_session.ts
import type { ClientBase } from "../../../base";
import type { CustomerCustomerSession, PortalAuthenticatedUser } from "../../models";

export const introspectCustomerSession = (client: ClientBase) => {
  /**
   * Introspect the current session and return its information.
   *
   * **Scopes**: `customer_portal:read` `customer_portal:write`
   *
   * @returns {CustomerCustomerSession}
   * @throws {PolarNetworkError} When a network error occurs
   * @throws {PolarRateLimitError} When the rate limit is exceeded
   * @throws {PolarServerError} When the server returns a 5xx error
   */
  return async (): Promise<CustomerCustomerSession> => {
    const pathParams = {};
    const queryParams = {};
    const request = client.buildRequest(
      "GET",
      "/v1/customer-portal/customer-session/introspect",
      pathParams,
      queryParams,
      undefined,
    );
    const response = await client.sendRequest(request);
    return client.parseResponse<CustomerCustomerSession>(response, "json", {});
  };
};
export const getAuthenticatedUserCustomerSession = (client: ClientBase) => {
  /**
   * Get information about the currently authenticated portal user.
   *
   * **Scopes**: `customer_portal:read` `customer_portal:write`
   *
   * @returns {PortalAuthenticatedUser}
   * @throws {PolarNetworkError} When a network error occurs
   * @throws {PolarRateLimitError} When the rate limit is exceeded
   * @throws {PolarServerError} When the server returns a 5xx error
   */
  return async (): Promise<PortalAuthenticatedUser> => {
    const pathParams = {};
    const queryParams = {};
    const request = client.buildRequest(
      "GET",
      "/v1/customer-portal/customer-session/user",
      pathParams,
      queryParams,
      undefined,
    );
    const response = await client.sendRequest(request);
    return client.parseResponse<PortalAuthenticatedUser>(response, "json", {});
  };
};

export function createCustomerSessionService(client: ClientBase) {
  return {
    introspect: introspectCustomerSession(client),
    getAuthenticatedUser: getAuthenticatedUserCustomerSession(client),
  };
}

export type CustomerSession = ReturnType<typeof createCustomerSessionService>;

File: sdk/python/polar/v2026_10/services/customer_portal/customer_session.py
from __future__ import annotations

from polar.base import (
    AsyncServiceBase,
    RequestTimeout,
    SyncServiceBase,
    parse_response_json,
)
from polar.v2026_10.outputs import (
    CustomerCustomerSession,
    PortalAuthenticatedUser,
)


class CustomerSessionSync(SyncServiceBase):
    def introspect(
        self,
        *,
        request_timeout: RequestTimeout | None = None,
        request_access_token: str | None = None,
    ) -> CustomerCustomerSession:
        """
        Introspect the current session and return its information.

        **Scopes**: `customer_portal:read` `customer_portal:write`

        Args:
            request_timeout: Timeout override for this request, in seconds or as an httpx.Timeout instance.
            request_access_token: Access token override for this request.


        Raises:
            PolarNetworkError: Raised when a network error occurs while making the request.
            PolarRateLimitError: Raised when the rate limit is exceeded.
            PolarServerError: Raised when the server returns a 5xx error response.
        """
        request = self.client.build_request(
            method="GET",
            url="/v1/customer-portal/customer-session/introspect",
            path_params={},
            query_params={},
            request_timeout=request_timeout,
            request_access_token=request_access_token,
        )
        response = self.client.send_request(request)
        return parse_response_json(response, CustomerCustomerSession)

    def get_authenticated_user(
        self,
        *,
        request_timeout: RequestTimeout | None = None,
        request_access_token: str | None = None,
    ) -> PortalAuthenticatedUser:
        """
        Get information about the currently authenticated portal user.

        **Scopes**: `customer_portal:read` `customer_portal:write`

        Args:
            request_timeout: Timeout override for this request, in seconds or as an httpx.Timeout instance.
            request_access_token: Access token override for this request.


        Raises:
            PolarNetworkError: Raised when a network error occurs while making the request.
            PolarRateLimitError: Raised when the rate limit is exceeded.
            PolarServerError: Raised when the server returns a 5xx error response.
        """
        request = self.client.build_request(
            method="GET",
            url="/v1/customer-portal/customer-session/user",
            path_params={},
            query_params={},
            request_timeout=request_timeout,
            request_access_token=request_access_token,
        )
        response = self.client.send_request(request)
        return parse_response_json(response, PortalAuthenticatedUser)


class CustomerSessionAsync(AsyncServiceBase):
    async def introspect(
        self,
        *,
        request_timeout: RequestTimeout | None = None,
        request_access_token: str | None = None,
    ) -> CustomerCustomerSession:
        """
        Introspect the current session and return its information.

        **Scopes**: `customer_portal:read` `customer_portal:write`

        Args:
            request_timeout: Timeout override for this request, in seconds or as an httpx.Timeout instance.
            request_access_token: Access token override for this request.


        Raises:
            PolarNetworkError: Raised when a network error occurs while making the request.
            PolarRateLimitError: Raised when the rate limit is exceeded.
            PolarServerError: Raised when the server returns a 5xx error response.
        """
        request = self.client.build_request(
            method="GET",
            url="/v1/customer-portal/customer-session/introspect",
            path_params={},
            query_params={},
            request_timeout=request_timeout,
            request_access_token=request_access_token,
        )
        response = await self.client.send_request(request)
        return parse_response_json(response, CustomerCustomerSession)

    async def get_authenticated_user(
        self,
        *,
        request_timeout: RequestTimeout | None = None,
        request_access_token: str | None = None,
    ) -> PortalAuthenticatedUser:
        """
        Get information about the currently authenticated portal user.

        **Scopes**: `customer_portal:read` `customer_portal:write`

        Args:
            request_timeout: Timeout override for this request, in seconds or as an httpx.Timeout instance.
            request_access_token: Access token override for this request.


        Raises:
            PolarNetworkError: Raised when a network error occurs while making the request.
            PolarRateLimitError: Raised when the rate limit is exceeded.
            PolarServerError: Raised when the server returns a 5xx error response.
        """
        request = self.client.build_request(
            method="GET",
            url="/v1/customer-portal/customer-session/user",
            path_params={},
            query_params={},
            request_timeout=request_timeout,
            request_access_token=request_access_token,
        )
        response = await self.client.send_request(request)
        return parse_response_json(response, PortalAuthenticatedUser)
